sunpy/net/dataretriever/client.py

```diff
@@ -89,10 +88,17 @@ class GenericClient(BaseClient):
     """
 
     @classmethod
-    def _get_patterns_for_attrs(cls, **kwargs):
+    def pre_hook(cls, *args, **kwargs):
```

Comment:
What are the args here? Does this need to be a full `*args, **kwargs`?

I think the query needs to be passed in here? Does anything else?

Reply:
Yes, `*args` and `**kwargs` of query shall be used for translation here. This pre-hook will be overwritten for some clients like `norh`. Where wavelength shall translate to either `tcz` or `tca` in the url.